src/App.tsx

```diff
@@ -1,14 +1,35 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [todosFromServer, setTodosFromServer] = useState<Todo[]>();
+  const [todos, setTodos] = useState<Todo[]>();
+  const [isLoading, setIsLoading] = useState(true);
+  const [isModal, setIsModal] = useState(-1);
```

Comment:
Consider initializing `isModal` with `null` or a similar value to indicate no modal is open. Using `-1` might be misleading as it suggests an invalid ID.